pypxe/dhcp.py

```diff
@@ -37,30 +37,43 @@ def __init__(self, **serverSettings):
         self.magic = struct.pack('!I', 0x63825363) #magic cookie
 
+        if self.logger == None:
+            import logging
+            import logging.handlers
+            # setup logger
+            self.logger = logging.getLogger("dhcp")
+            handler = logging.StreamHandler()
+            formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+            handler.setFormatter(formatter)
+            self.logger.addHandler(handler)
+
+            if self.mode_debug:
+                self.logger.setLevel(logging.DEBUG)
+
         if self.http and not self.ipxe:
-            print '\nWARNING: HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.\n'
+            self.logger.warning('HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.')
         if self.ipxe and self.http:
             self.filename = 'http://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
         if self.ipxe and not self.http:
             self.filename = 'tftp://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
 
-        if self.mode_debug:
-            print 'NOTICE: DHCP server started in debug mode. DHCP server is using the following:'
-            print '\tDHCP Server IP: {}'.format(self.ip)
-            print '\tDHCP Server Port: {}'.format(self.port)
-            print '\tDHCP Lease Range: {} - {}'.format(self.offerfrom, self.offerto)
-            print '\tDHCP Subnet Mask: {}'.format(self.subnetmask)
-            print '\tDHCP Router: {}'.format(self.router)
-            print '\tDHCP DNS Server: {}'.format(self.dnsserver)
-            print '\tDHCP Broadcast Address: {}'.format(self.broadcast)
-            print '\tDHCP File Server IP: {}'.format(self.fileserver)
-            print '\tDHCP File Name: {}'.format(self.filename)
-            print '\tProxyDHCP Mode: {}'.format(self.mode_proxy)
-            print '\tUsing iPXE: {}'.format(self.ipxe)
-            print '\tUsing HTTP Server: {}'.format(self.http)
+        self.logger.debug('NOTICE: DHCP server started in debug mode. DHCP server is using the following:')
+        self.logger.debug('\tDHCP Server IP: ' + self.ip)
+        self.logger.debug('\tDHCP Server Port: ' + str (self.port))
+        self.logger.debug('\tDHCP Lease Range: ' + self.offerfrom + ' - ' + self.offerto)
+        self.logger.debug('\tDHCP Subnet Mask: ' + self.subnetmask)
+        self.logger.debug('\tDHCP Router: ' + self.router)
+        self.logger.debug('\tDHCP DNS Server: ' + self.dnsserver)
+        self.logger.debug('\tDHCP Broadcast Address: ' + self.broadcast)
+        self.logger.debug('\tDHCP File Server IP: ' + self.fileserver)
+        self.logger.debug('\tDHCP File Name: ' + self.filename)
+        self.logger.debug('\tProxyDHCP Mode: ' + str(self.mode_proxy))
+        self.logger.debug('\tUsing iPXE: ' + str(self.ipxe))
+        self.logger.debug('\tUsing HTTP Server: ' + str(self.http))
```

Comment:
In L65-L76 should we remove the `\t` characters since we're using the logger now? If using the logger perhaps we shouldn't use the `\t`? Thoughts?